cmd/agentbasedinstaller/register.go

```diff
@@ -85,21 +85,25 @@ func RegisterInfraEnv(ctx context.Context, log *log.Logger, bmInventory *client.
 		return nil, infraenvErr
 	}
 
+	infraEnvParams := controllers.CreateInfraEnvParams(&infraEnv, models.ImageType(imageTypeISO), pullSecret, modelsCluster.ID, modelsCluster.OpenshiftVersion)
+
 	var nmStateConfig aiv1beta1.NMStateConfig
-	if nmStateErr := getFileData(nmStateConfigPath, &nmStateConfig); nmStateErr != nil {
-		return nil, nmStateErr
-	}
 
-	staticNetworkConfig, processErr := processNMStateConfig(log, infraEnv, nmStateConfig)
-	if processErr != nil {
-		return nil, processErr
-	}
+	fileInfo, _ := os.Stat(nmStateConfigPath)
```

Comment:
It would be nice to add a test for this new case, by extending the suite for the agentbasedinstaller (see [here](https://github.com/openshift/assisted-service/blob/3541ee5294a89fe9a10a03f1b561790a30a10cc2/cmd/agentbasedinstaller/agentbasedinstaller_suite_test.go#L50), there's the possibility to mock both the filesystem and the network)

Reply:
I would like to create a separate PR to add the test case for covering this scenario and to refactor the existing code.

Reply:
/lgtm